### PR TITLE
Add search highlighting and nav

### PR DESCRIPTION
**Pull request by joelochlann**
Adds a second control, non-editable, to navigate between main search highlights

https://user-images.githubusercontent.com/5122968/176723786-18bc15d2-95f2-4d48-a446-4d6f2895bac7.mov

### CODE REVIEW

File: backend/app/controllers/api/PagesController.scala [hunk 1]
```diff
@@ -29,7 +29,9 @@ class PagesController(val controllerComponents: AuthControllerComponents, manife
   // Get language and highlight data for a given page
   private def frontendPageFromQuery(uri: Uri, pageNumber: Int, username: String, sq: Option[String], fq: Option[String]): Attempt[FrontendPage] = {
     // Across documents
-    val searchQuery = sq.map(Chips.parseQueryString)
+    // TODO: wtf why do we get horrific error when parsing chips?
```
Comment (joelochlann) [inline backend/app/controllers/api/PagesController.scala:32]:
```bash
[ERROR] 2022-07-01 11:04:07 +0100 - utils.controller.FailureToResultMapper$ - pfi-akka.actor.default-dispatcher-5 - Unknown Failure: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'scala': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (String)"scala"; line: 1, column: 6]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1851)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:717)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:2898)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:1944)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:776)
	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4662)
	at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:4482)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2728)
	at play.api.libs.json.jackson.JacksonJson$.parseJsValue(JacksonJson.scala:285)
	at play.api.libs.json.StaticBinding$.parseJsValue(StaticBinding.scala:17)
	at play.api.libs.json.Json$.parse(Json.scala:173)
	at model.frontend.Chips$.parseQueryString(Chip.scala:55)
	at controllers.api.PagesController.$anonfun$frontendPageFromQuery$1(PagesController.scala:33)
	at scala.Option.map(Option.scala:230)
	at controllers.api.PagesController.frontendPageFromQuery(PagesController.scala:33)
	at controllers.api.PagesController.$anonfun$findInDocument$5(PagesController.scala:135)
	at controllers.api.PagesController.$anonfun$findInDocument$5$adapted(PagesController.scala:135)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:38)
	at scala.collection.TraversableLike.map(TraversableLike.scala:286)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at controllers.api.PagesController.$anonfun$findInDocument$4(PagesController.scala:135)
	at utils.attempt.Attempt.$anonfun$flatMap$1(Attempt.scala:26)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:56)
	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:93)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:93)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:48)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:48)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

Comment (itsibitzi) [reply backend/app/controllers/api/PagesController.scala:32]: Usually leave it to Elasticsearch to figure out, though tbh I don't think anything other than a plain string makes much sense here.

**Comment by itsibitzi** [comment]
Looking  good! Could possibly remove the extra set of rotate buttons? :)

File: backend/app/controllers/api/PagesController.scala (hunk 1 above)
Comment (joelochlann) [reply backend/app/controllers/api/PagesController.scala:32]: Well, the text viewer and the page viewer (mark 1) both handled chip syntax in their `q` param so clearly we should. I will dig through that code and figure out how it works.

Just uncommenting this and passing, e.g. `sq=%5B"socialism"%5D` doesn't work as it seems to pass it with brackets to elasticsearch which blows up

**Comment by joelochlann** [comment]
@itsibitzi have removed buttons. I think the input should probably go to the left edge but end of Friday I'm somehow finding it inordinately difficult to style so maybe that could be tweaked in subsequent PR, if important.

also the chips rendering thing I'd like to address after doing the wiring up to search results (behind feature switch)

Screenshot 2022-07-01 at 17 49 17

**Comment by joelochlann** [comment]
renders chips now

Screenshot 2022-07-04 at 11 54 37

**Comment by joelochlann** [comment]
>  should probably go to the left edge

ahh I see the misreading... yeah this might be better, but will need to see with sidebar. potentially they could be side-by-side at the top of the document, displacing it vertically (so they never cover it)

**Review by itsibitzi — APPROVED** [review]

**Pull request merged**